tests/unit/models/gpflux/test_models.py

```diff
@@ -0,0 +1,298 @@
+
+    with pytest.raises(ValueError):
+        DeepGaussianProcess(dgp, optimizer=optimizer)
+
+
+def test_dgp_raises_for_latent_variable_layer() -> None:
+    num_data = 5
+
+    encoder = gpflux.encoders.DirectlyParameterizedNormalDiag(num_data, 1)
+    prior = tfp.distributions.MultivariateNormalDiag(np.zeros(1), np.ones(1))
+    lv = gpflux.layers.LatentVariableLayer(prior, encoder)
+    kernel = gpflow.kernels.SquaredExponential(lengthscales=[1.0] * 2)
+    num_inducing = 5
+    inducing_variable = gpflow.inducing_variables.InducingPoints(
+        np.concatenate(
+            [
+                np.linspace(-1, 1, num_inducing).reshape(-1, 1),
+                np.random.randn(num_inducing, 1),
+            ],
+            axis=1,
+        )
+    )
+    gp_layer = gpflux.layers.GPLayer(
+        kernel,
+        inducing_variable,
+        num_data=num_data,
+        num_latent_gps=1,
+        mean_function=gpflow.mean_functions.Zero(),
+    )
+
+    likelihood_layer = gpflux.layers.LikelihoodLayer(gpflow.likelihoods.Gaussian(0.01))
+
+    dgp = DeepGP([lv, gp_layer], likelihood_layer)
+
+    with pytest.raises(ValueError):
+        DeepGaussianProcess(dgp)
+
+
+def test_dgp_raises_for_keras_layer() -> None:
+    keras_layer_1 = tf.keras.layers.Dense(50, activation="relu")
+    keras_layer_2 = tf.keras.layers.Dense(2, activation="relu")
+
+    kernel = gpflow.kernels.SquaredExponential()
+    num_inducing = 5
+    inducing_variable = gpflow.inducing_variables.InducingPoints(
+        np.concatenate(
+            [
+                np.random.randn(num_inducing, 2),
+            ],
+            axis=1,
+        )
+    )
+    gp_layer = gpflux.layers.GPLayer(
+        kernel,
+        inducing_variable,
+        num_data=5,
+        num_latent_gps=1,
+        mean_function=gpflow.mean_functions.Zero(),
+    )
+
+    likelihood_layer = gpflux.layers.LikelihoodLayer(gpflow.likelihoods.Gaussian(0.01))
+
+    dgp = DeepGP([keras_layer_1, keras_layer_2, gp_layer], likelihood_layer)
+
+    with pytest.raises(ValueError):
+        DeepGaussianProcess(dgp)
+
+
+def test_dgp_warns_for_whitened_layers() -> None:
+    num_data = 5
+    num_inducing = num_data
+
+    Z = np.linspace(-1, 1, num_inducing).reshape(-1, 1)
+    kernel_1 = gpflow.kernels.SquaredExponential()
+    inducing_variable_1 = gpflow.inducing_variables.InducingPoints(Z.copy())
+    gp_layer_1 = gpflux.layers.GPLayer(
+        kernel_1,
+        inducing_variable_1,
+        num_data=num_data,
+        num_latent_gps=1,
+        whiten=False,
+    )
+
+    kernel_2 = gpflow.kernels.SquaredExponential()
+    inducing_variable_2 = gpflow.inducing_variables.InducingPoints(Z.copy())
+    gp_layer_2 = gpflux.layers.GPLayer(
+        kernel_2,
+        inducing_variable_2,
+        num_data=num_data,
+        num_latent_gps=1,
+        whiten=True,
+    )
+
+    dgp = DeepGP([gp_layer_1, gp_layer_2], gpflow.likelihoods.Gaussian(0.01))
+
+    with pytest.warns(UserWarning):
```

Comment:
this recognizes there was a warning, but not what the warning was, right?
you can check for what is contained in the warning message and check if some keyword is present in that string

Reply:
Yes, that's true, but at the moment we don't have any other `UserWarning` s in the code so hopefully it should be ok? But probably best to be safe about it

Reply:
not at the moment, no, but its easy to add another and forget that this test would then potentially incorrectly pass

Reply:
Done